pkg/query/flamegraph_arrow.go

```diff
@@ -528,6 +528,104 @@ func (fb *flamegraphBuilder) mergeSymbolizedRows(
 			fb.intersectLabels(r, t, recordLabelIndex, sampleIndex, cr)
 		}
 
+		// Compare the existing row's metadata values with the one we're merging.
+		// If these values differ we need to set the row's metadata column to null.
+		{
+			if !fb.builderMappingFileIndices.IsNull(cr) {
```

Comment:
I realize this seems drastic, but it has been shown that the negation here is a perf problem, so use `IsValid` rather than `IsNull`, they actually do different validity bitmap checks and therefore we won't loose on perf.

Reply:
Right. I can add those later on. However, right now the underlying builders don't have those methods and I instead opted for `!IsNull`. Making this consistent with `IsValid` should be a follow-up.

Reply:
Ah got it, ok lgtm then let's move forward